addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRuleUnitTest.java

```diff
@@ -105,6 +112,18 @@ void shouldReturnExpectedMappings() {
                 is(equalTo(CommonAlertTag.WSTG_V42_SESS_05_CSRF.getValue())));
     }
 
+    @Test
+    void shouldNotRaiseAlertIfContentTypeIsNotHTML() {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.getResponseHeader().addHeader(HttpHeader.CONTENT_TYPE, "application/json");
+        msg.setResponseBody("no html");
```

Comment:
The content should actually lead to an alert, otherwise this is not testing the behaviour we want (i.e. the test still passes even without the condition added by this PR).

Reply:
Extracting a method from one of the existing cases that raise an alert is probably the best, so they are kept in sync.

Reply:
What would be a good signature for this method? Should we keeping using the `msg` instance variable and have something like `String getMessageBodyVunerableToCSRF()`?
Or should we create a message from scratch like in [ascanrulesBeta/CsrfTokenScanRuleUnitTest](https://github.com/zaproxy/zap-extensions/blob/main/addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRuleUnitTest.java#L364)?

Reply:
`void formWithMissingAntiCsrfToken()` with https://github.com/zaproxy/zap-extensions/blob/15bb4bff1e27710a9f9d90d0153a1188dcfa25bc/addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRuleUnitTest.java#L142-L143